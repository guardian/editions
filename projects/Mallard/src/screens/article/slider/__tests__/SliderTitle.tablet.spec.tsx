import { Dimensions } from 'react-native';
import { baseTests } from './SliderTitle-BaseTests';

jest.mock('react-native-device-info', () => ({
	isTablet: () => true,
}));

jest.spyOn(Dimensions, 'get').mockReturnValue({
	width: 700,
	height: 100,
	scale: 1,
	fontScale: 1,
});

baseTests('SliderTitle - iOS - Tablet');
