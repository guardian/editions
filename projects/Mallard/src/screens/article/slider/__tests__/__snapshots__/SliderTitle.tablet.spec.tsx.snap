// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SliderTitle - iOS - Tablet should display a Front SliderTitle with the correct styles 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "maxWidth": undefined,
      "paddingLeft": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 28,
          "lineHeight": 32,
        }
      }
    >
      Top Stories
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 28,
          "lineHeight": 32,
        }
      }
    >
       
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingVertical": 2,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, 1)",
          "borderRadius": 7,
          "height": 14,
          "marginRight": 7,
          "width": 14,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, 1)",
          "borderRadius": 7,
          "height": 14,
          "marginRight": 7,
          "width": 14,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, 1)",
          "borderRadius": 7,
          "height": 14,
          "marginRight": 7,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`SliderTitle - iOS - Tablet should display an Article SliderTitle with a subtitle different to the title 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "maxWidth": 740,
      "paddingLeft": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 20,
          "lineHeight": 22,
        }
      }
    >
      Top Stories
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 20,
          "lineHeight": 22,
        }
      }
    >
       
      Scary News
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingVertical": 2,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
  </View>
</View>
`;

exports[`SliderTitle - iOS - Tablet should display an Article SliderTitle with a subtitle the same as the title 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "maxWidth": 740,
      "paddingLeft": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 20,
          "lineHeight": 22,
        }
      }
    >
      Top Stories
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingVertical": 2,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
  </View>
</View>
`;

exports[`SliderTitle - iOS - Tablet should display an Article SliderTitle with the correct styles 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "maxWidth": 740,
      "paddingLeft": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 20,
          "lineHeight": 22,
        }
      }
    >
      Top Stories
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 20,
          "lineHeight": 22,
        }
      }
    >
       
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingVertical": 2,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
  </View>
</View>
`;

exports[`SliderTitle - iOS - Tablet should display an Front SliderTitle with a subtitle different to the title 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "maxWidth": undefined,
      "paddingLeft": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 28,
          "lineHeight": 32,
        }
      }
    >
      Top Stories
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 28,
          "lineHeight": 32,
        }
      }
    >
       
      Scary News
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingVertical": 2,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, 1)",
          "borderRadius": 7,
          "height": 14,
          "marginRight": 7,
          "width": 14,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, 1)",
          "borderRadius": 7,
          "height": 14,
          "marginRight": 7,
          "width": 14,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, 1)",
          "borderRadius": 7,
          "height": 14,
          "marginRight": 7,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`SliderTitle - iOS - Tablet should display an Front SliderTitle with a subtitle the same as the title 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "maxWidth": undefined,
      "paddingLeft": 10,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 28,
          "lineHeight": 32,
        }
      }
    >
      Top Stories
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingVertical": 2,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, 1)",
          "borderRadius": 7,
          "height": 14,
          "marginRight": 7,
          "width": 14,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, 1)",
          "borderRadius": 7,
          "height": 14,
          "marginRight": 7,
          "width": 14,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, 1)",
          "borderRadius": 7,
          "height": 14,
          "marginRight": 7,
          "width": 14,
        }
      }
    />
  </View>
</View>
`;

exports[`SliderTitle - iOS - Tablet should display an default SliderTitle with a prescribed start index 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "maxWidth": 740,
      "paddingLeft": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 20,
          "lineHeight": 22,
        }
      }
    >
      Top Stories
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 20,
          "lineHeight": 22,
        }
      }
    >
       
    </Text>
  </View>
  <View
    style={
      Object {
        "flexDirection": "row",
        "paddingVertical": 2,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#DCDCDC",
          "borderRadius": 4,
          "height": 8,
          "marginRight": 2,
          "width": 8,
        }
      }
    />
  </View>
</View>
`;

exports[`SliderTitle - iOS - Tablet should display no dots when numOfItems is 1 or less 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "maxWidth": 740,
      "paddingLeft": 0,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#000000",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 20,
          "lineHeight": 22,
        }
      }
    >
      Top Stories
    </Text>
    <Text
      style={
        Object {
          "color": "grey",
          "fontFamily": "GTGuardianTitlepiece-Bold",
          "fontSize": 20,
          "lineHeight": 22,
        }
      }
    >
       
    </Text>
  </View>
</View>
`;
